### PR TITLE
feat(metrics): Add metrics for deleteSnapshot operation

### PR DESCRIPTION
**Pull request by aravindmd**
Adding metrics for deleteSnaphot operation

### CODE REVIEW

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/converters/DeleteAmazonSnapshotAtomicOperationConverter.java [hunk 1]
```diff
@@ -24,16 +24,24 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
-
+import com.netflix.spectator.api.Registry;
 import java.util.Map;
 
 @AmazonOperation(AtomicOperations.DELETE_SNAPSHOT)
 @Component
 public class DeleteAmazonSnapshotAtomicOperationConverter extends AbstractAtomicOperationsCredentialsSupport {
+  private Registry registry;
```
Comment (marchello2000) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/converters/DeleteAmazonSnapshotAtomicOperationConverter.java]: you can make it `final`

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java [hunk 2]
```diff
@@ -47,9 +57,20 @@ private static Task getTask() {
...
+        .getAmazonEC2(description.getCredentials(), description.getRegion())
+        .deleteSnapshot(new DeleteSnapshotRequest().withSnapshotId(description.getSnapshotId()));
+    } catch (AmazonEC2Exception ec2Exception) {
+      registry.counter(deleteSnapshotTaskId.withTag("success", false)).increment();
+      log.error("Failed to delete snapshotId with exception: {}", ec2Exception.getMessage());
+      throw ec2Exception;
+    } catch (Exception e) {
+      registry.counter(deleteSnapshotTaskId.withTag("success", false)).increment();
+      log.error("Failed to delete snapshotId with exception: {}", e.getMessage());
+      throw e;
```
Comment (marchello2000) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java]: i don't think there is anything diff in these two handlers, did you intend to have some different behavior here? if not, can just leave the `Exception` one

Comment (ajordens) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java]: Ditto

Comment (aravindmd) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java]: agree , updated it to just handle ```Exception```

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/converters/DeleteAmazonSnapshotAtomicOperationConverter.java (hunk 1 above)
Comment (aravindmd) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/converters/DeleteAmazonSnapshotAtomicOperationConverter.java]: done

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java (hunk 2 above)
Comment (marchello2000) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java]: I should've noted this earlier... Might be useful to log the `snapshotId` to the `log.error`

Comment (aravindmd) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java]: the exception message has the details of the ```snapshotId``` associated ```ami``` etc

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java [hunk 3]
```diff
@@ -47,9 +57,16 @@ private static Task getTask() {
...
-    amazonClientProvider
-      .getAmazonEC2(description.getCredentials(), description.getRegion())
-      .deleteSnapshot(new DeleteSnapshotRequest().withSnapshotId(description.getSnapshotId()));
+    try {
+      amazonClientProvider
+        .getAmazonEC2(description.getCredentials(), description.getRegion())
+        .deleteSnapshot(new DeleteSnapshotRequest().withSnapshotId(description.getSnapshotId()));
+    } catch (Exception e) {
+      registry.counter(deleteSnapshotTaskId.withTag("success", false)).increment();
+      log.error("Failed to delete snapshotId with exception: {}", e.getMessage());
```
Comment (robzienert) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java]: I'd prefer:

`log.error(format("Failed to delete snapshotId %s", description.getSnapshotId()), e);`

This way we actually get a stacktrace, and we get real context of _what_ snapshot failed. In the log's current form, we wouldn't have much actionable and may have a more challenging time sourcing the location / origin of this error.

**Pull request merged**